packages/nice-grpc/README.md

```diff
@@ -994,6 +994,14 @@ const client2 = clientFactory.use(middlewareC).create(Service2, channel2);
 In the above example, `Service1` client gets `middlewareA` and `middlewareB`,
 and `Service2` client gets `middlewareA` and `middlewareC`.
 
+If needed, client types can be augmented with the `CallOptionsExt` generic parameter:
+
+```ts
+const client: ExampleServiceClient<{callOption?: number}> = createClientFactory()
+  .use(middleware)
+  .create(ExampleService, channel);
+```
```

Comment:
This may be confusing, since the augmentation is done by the middleware, not the `Client` type.

This code example shows how to correctly annotate `client` type given that `middleware` has type `ClientMiddleware<{callOption?: number}>`.

Reply:
I have updated the description before the example to hopefully be more clear. Even though it's intended to not need manually reference a `Client` type, my use case _was_ doing that before I added a middleware & this was a real hang up for me because I didn't dig deep enough into the `createClientFactory()`.

It may be that this is a me-specific problem, if you think so feel free to close this PR; the docs can't spell out every stumbling block & edge case.

Reply:
No, there are many cases where you want to specify the `Client` type. For example when you pass the client as an argument. So it's a nice addition to the docs.

Thanks!